bioimageio/core/prediction_pipeline/_processing.py

```diff
@@ -0,0 +1,228 @@
+from dataclasses import InitVar, dataclass, field
+from itertools import chain
+from typing import (
+    Callable,
+    ClassVar,
+    Dict,
+    FrozenSet,
+    List,
+    Literal,
+    Optional,
+    Sequence,
+    Set,
+    Tuple,
+    Type,
+    Union,
+    get_args,
+)
+
+import numpy as np
+import xarray as xr
+
+from bioimageio.core.resource_io import nodes
+from ._types import Transform
+from .statistical_measures import Mean, Measure, Percentile, Std
+from bioimageio.spec.model.v0_3.raw_nodes import PreprocessingName
+
+
+def ensure_dtype(tensor: xr.DataArray, *, dtype) -> xr.DataArray:
+    """
+    Convert array to a given datatype
+    """
+    return tensor.astype(dtype)
+
+
+@dataclass
+class Processing:
+    apply_to: str
```

Comment:
not all preprocessings use the `apply_to` parameter.

Reply:
they have to (should)

Reply:
ah, here.. `apply_to` has been, renamed to `tensor_name`